### PR TITLE
[HINTS] Math cmp is le felt

### PR DESCRIPTION
**Pull request by Juan-M-V**
# Math cmp is le felt

## Description

Implements is_le_felt

## Checklist
- [x] Unit tests added
- [x] Integration tests added.

### CODE REVIEW

File: src/types/relocatable.rs [hunk 1]
```diff
@@ -406,6 +414,17 @@ mod tests {
...
 
+    #[test]
+    fn mod_floor_int() {
+        let num = MaybeRelocatable::Int(bigint!(7));
+        let div = bigint!(5);
+        let expected_rem = MaybeRelocatable::Int(bigint!(2));
+        assert_eq!(
+            num.mod_floor(&div).expect("Unexpected mod floor fail"),
+            expected_rem
+        );
```
Comment (Oppen) [inline src/types/relocatable.rs:425]:
```suggestion
        assert_eq!(num.mod_floor(&div), Some(expected_rem));
```

File: src/vm/hints/execute_hint.rs [hunk 2]
```diff
@@ -492,6 +494,163 @@ mod tests {
...
+        vm.run_context.ap = MaybeRelocatable::from((1, 0));
+        vm.run_context.fp = MaybeRelocatable::from((0, 2));
+        //Insert ids into memory
+        //ids.a
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((0, 0)),
+                &MaybeRelocatable::from(bigint!(1)),
+            )
+            .expect("Unexpected memroy insert fail");
```
Comment (Oppen) [inline src/vm/hints/execute_hint.rs:520]:
```suggestion
            .expect("Unexpected memory insert fail");
```

File: src/vm/hints/execute_hint.rs [hunk 3]
```diff
@@ -492,6 +494,163 @@ mod tests {
...
+                register: Register::FP,
+                offset: -2,
+            },
+            HintReference {
+                register: Register::FP,
+                offset: -1,
+            },
+        ];
+        //Execute the hint
+        assert_eq!(execute_hint(&mut vm, hint_code, ids), Ok(()));
```
Comment (Oppen) [inline src/vm/hints/execute_hint.rs:544]:
```suggestion
        assert!(execute_hint(&mut vm, hint_code, ids).is_ok());
```

File: src/vm/hints/hint_utils.rs [hunk 4]
```diff
@@ -280,6 +280,81 @@ pub fn assert_le_felt(
...
+    ) {
+        (a_addr, b_addr)
+    } else {
+        return Err(VirtualMachineError::FailedToGetIds);
+    };
+    match (vm.memory.get(&a_addr), vm.memory.get(&b_addr)) {
+        (Ok(Some(maybe_rel_a)), Ok(Some(maybe_rel_b))) => {
+            for (name, builtin) in &vm.builtin_runners {
+                //Check that range_check_builtin is present
+                if name == &String::from("range_check") {
```
Comment (Oppen) [inline src/vm/hints/hint_utils.rs:313]:
```suggestion
                if name != &String::from("range_check") {
                    continue;
                }
```
Then reduce an indentation level of everything inside.

File: src/vm/hints/hint_utils.rs [hunk 5]
```diff
@@ -280,6 +280,81 @@ pub fn assert_le_felt(
...
+                    match builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>() {
+                        None => return Err(VirtualMachineError::NoRangeCheckBuiltin),
```
Comment (Oppen) [inline src/vm/hints/hint_utils.rs:315]:
```rust
let _builtin = if let ... {
} else {
    return Err(...);
}

File: src/vm/hints/hint_utils.rs (hunk 4 above)
Comment (Oppen) [reply src/vm/hints/hint_utils.rs:313]: Alternatively we could break with the answer and work outside of the loop when found.

File: src/vm/hints/hint_utils.rs [hunk 6]
```diff
@@ -280,6 +280,81 @@ pub fn assert_le_felt(
...
+                                Ok(_) => return Ok(()),
+                                Err(memory_error) => {
+                                    return Err(VirtualMachineError::MemoryError(memory_error))
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            Err(VirtualMachineError::NoRangeCheckBuiltin)
```
[the remark this answers is not in the document]
Comment (Juan-M-V) [reply src/vm/hints/hint_utils.rs]: Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/270/commits/f3488c01657215f6ca0e8bec8ba077aebeb161b4

File: src/vm/hints/hint_utils.rs [hunk 7]
```diff
@@ -311,39 +311,32 @@ pub fn is_le_felt(
...
+                        }
+                        match vm
+                            .memory
+                            .insert(&vm.run_context.ap, &MaybeRelocatable::from(value))
+                        {
+                            Ok(_) => return Ok(()),
+                            Err(memory_error) => {
+                                return Err(VirtualMachineError::MemoryError(memory_error))
                             }
                         }
```
Comment (Oppen) [inline src/vm/hints/hint_utils.rs:341]:
```suggestion
                        match vm
                            .memory
                            .insert(&vm.run_context.ap, &MaybeRelocatable::from(value))
                            .map_err(VirtualMachineError::MemoryError)
```

Comment (Juan-M-V) [reply src/vm/hints/hint_utils.rs]: Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/270/commits/cc7224046e1adc0c63542217fd0b25400674e981

**Review by Oppen — APPROVED** [review]

**Pull request merged**